pkg/apis/sensor/v1alpha1/types.go

```diff
@@ -249,6 +249,11 @@ type DataFilter struct {
 	// Can be ">=", ">", "=", "!=", "<", or "<=".
 	// Is optional, and if left blank treated as equality "=".
 	Comparator Comparator `json:"comparator,omitempty" protobuf:"bytes,4,opt,name=comparator,casttype=Comparator"`
+	// DataTemplate is a go-template for extracting a string from the event's data.
```

Comment:
Let's name it `template` directly, which means `template function`.

This `template` will work together with `path`, it uses the result of `path` as its input, for example, spec below means lower case of `body.value`.

```
            filters:
            data:
              - path: body.value
                 template: {{ .Input | lower }}
```

The template should work not only string, but types like number, it is used to additional processing for the value of `path` before filtering.